elliottlib/cli/find_builds_cli.py

```diff
@@ -91,33 +87,42 @@ def find_builds_cli(runtime, advisory, default_advisory_type, builds, kind, from
         raise click.BadParameter('Use only one of --use-default-advisory or --attach')
 
     runtime.initialize(mode='images' if kind == 'image' else 'none')
-    base_tag, product_version, tag_pv_map = _determine_errata_info(runtime)
+    base_tag, tag_pv_map = _determine_errata_info(runtime)
 
     if default_advisory_type is not None:
         advisory = find_default_advisory(runtime, default_advisory_type)
 
     ensure_erratatool_auth()  # before we waste time looking up builds we can't process
 
     # get the builds we want to add
-    unshipped_builds = []
-    session = requests.Session()
+    unshipped_nvrs = []
     if builds:
-        unshipped_builds = _fetch_builds_by_id(builds, product_version, session)
+        green_prefix('Build NVRs provided: ')
+        click.echo('Manually verifying the builds exist')
+        unshipped_nvrs = builds
     elif from_diff:
-        unshipped_builds = _fetch_builds_from_diff(from_diff[0], from_diff[1], product_version, session)
+        green_print('Fetching changed images between payloads...')
+        unshipped_nvrs = elliottlib.openshiftclient.get_build_list(from_diff[0], from_diff[1])
     else:
         if kind == 'image':
-            unshipped_builds = _fetch_builds_by_kind_image(runtime, tag_pv_map, session)
+            unshipped_nvrs = _fetch_builds_by_kind_image(runtime, tag_pv_map)
```

Comment:
Converting build objects to nvr strings then fetch building objects again is duplicate effort and time consuming.